src/main/cluster-detectors/distribution-detector.ts

```diff
@@ -4,6 +4,7 @@ import { ClusterMetadataKey } from "../cluster";
 export class DistributionDetector extends BaseClusterDetector {
   key = ClusterMetadataKey.DISTRIBUTION;
   version: string;
+  private cache = new Map<string, any>();
```

Comment:
Hmm.. it seems that each detect run creates a new instance from detector class so this cache does not help at all 🤔 /cc @nevalla

Reply:
It could if you make it static. But do we want to have a cache for detection?

Reply:
> It could if you make it static. But do we want to have a cache for detection?

Detector is per cluster which makes static a bit complex. We can have a cache per cluster because distribution does not really change ever.

Reply:
Is this to avoid making unnecessary api calls? Don't you have the cluster object and can check if the distribution is already set? Or we don't persist the distribution? Or maybe I need to study the code more deeply...

Reply:
@jim-docker yes, we don't want to hit k8s api unless really necessary. Idk how did I miss that this class gets `Cluster` object, I'll refactor 👍

Reply:
Oh no, using `Cluster` here is not enough, if end up with `unknown` we need to keep detecting -> we still need a separate cache. I'll refactor even more ...

Reply:
Decided to remove caching altogether after discussing with @nevalla . Detectors are triggered every 15 minutes which is not too often. We can improve this logic later.